### PR TITLE
Chain simulator test only processing node tests

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage for chain simulator
  
## Proposed changes
- added tests for testOnlyProcessingNode

## Testing procedure
- tests only

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by miiu96 — APPROVED** [review]

File: node/chainSimulator/components/testOnlyProcessingNode.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func NewTestOnlyProcessingNode(args ArgsTestOnlyProcessingNode) (*testOnlyProces
 		return nil, err
 	}
 
-	err = instance.createBroadcastMessanger()
+	err = instance.createBroadcastMessenger()
```
Comment (iulianpascalau) [inline node/chainSimulator/components/testOnlyProcessingNode.go:223]: :+1:

File: node/chainSimulator/components/testOnlyProcessingNode_test.go [hunk 2]
```diff
@@ -81,27 +85,410 @@ func TestNewTestOnlyProcessingNode(t *testing.T) {
...
-	args := createMockArgsTestOnlyProcessingNode(t)
-	node, err := NewTestOnlyProcessingNode(args)
-	require.Nil(t, err)
+func TestTestOnlyProcessingNode_SetKeyValueForAddress(t *testing.T) {
+	// TODO reinstate test after Wasm VM pointer fix
+	if testing.Short() {
+		t.Skip("cannot run with -race -short; requires Wasm VM fix")
+	}
+
+	t.Parallel()
```
Comment (iulianpascalau) [inline node/chainSimulator/components/testOnlyProcessingNode_test.go:160]: I would remove t.Parallel for tests involving chain simulator components that are actually doing something. They are pretty "heavy" components

Comment (sstanculeanu) [reply node/chainSimulator/components/testOnlyProcessingNode_test.go:160]: removed

**Review by iulianpascalau — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by ssd04 — DISMISSED** [review]

**Review by ssd04 — DISMISSED** [review]

File: node/chainSimulator/components/testOnlyProcessingNode_test.go [hunk 3]
```diff
@@ -152,11 +132,6 @@ func TestNewTestOnlyProcessingNode(t *testing.T) {
 }
 
 func TestTestOnlyProcessingNode_SetKeyValueForAddress(t *testing.T) {
-	// TODO reinstate test after Wasm VM pointer fix
```
Comment (iulianpascalau) [inline node/chainSimulator/components/testOnlyProcessingNode_test.go]: Please put this back

File: node/chainSimulator/components/testOnlyProcessingNode_test.go [hunk 4]
```diff
@@ -272,11 +239,6 @@ func TestTestOnlyProcessingNode_SetKeyValueForAddress(t *testing.T) {
 }
 
 func TestTestOnlyProcessingNode_SetStateForAddress(t *testing.T) {
-	// TODO reinstate test after Wasm VM pointer fix
```
Comment (iulianpascalau) [inline node/chainSimulator/components/testOnlyProcessingNode_test.go]: please put this back

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: node/chainSimulator/components/testOnlyProcessingNode_test.go (hunk 3 above)
Comment (sstanculeanu) [reply node/chainSimulator/components/testOnlyProcessingNode_test.go]: added back

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**